tests/unit/plugins/inventory/test_kubevirt.py

```diff
@@ -235,3 +420,724 @@ def test_ansible_connection_winrm(inventory, host_vars, client, vmi, expected):
+    result = inventory.get_host_from_service(request.getfixturevalue(service), node_name)
+    assert result == expected
+
+
+@pytest.fixture(scope="module")
+def load_balancer_port():
+    return {
+        "spec": {
+            "type": "LoadBalancer",
+            "ports": [{"port": "80"}],
+        },
+    }
+
+
+@pytest.fixture(scope="module")
+def node_port_port():
+    return {
+        "spec": {
+            "type": "NodePort",
+            "ports": [{"nodePort": "8080"}],
+        }
+    }
+
+
+@pytest.mark.parametrize(
+    "service,port",
+    [
+        ("service_other", None),
+        ("load_balancer_port", "80"),
+        ("node_port_port", "8080"),
+    ],
+)
+def test_port_from_service(request, inventory, service, port):
+    assert port == inventory.get_port_from_service(request.getfixturevalue(service))
+
+
+def test_parse(monkeypatch, inventory):
+    monkeypatch.setattr(inventory, "_read_config_data", lambda path: {"host_format": "default-test"})
+    monkeypatch.setattr(inventory, "_get_cache_prefix", lambda _: None)
+    monkeypatch.setattr(inventory, "setup", lambda a, b, c: None)
+
+    inventory.parse(inventory, None, "test")
+    assert inventory.host_format == "default-test"
+
+
+@pytest.fixture(scope="module")
+def connection_none():
+    return None
+
+
+@pytest.fixture(scope="module")
+def connection_not_list():
+    return "test"
+
+
+@pytest.fixture(scope="module")
+def connection_list_not_dict():
+    return [
+        "test",
+        "test"
+    ]
+
+
+@pytest.fixture(scope="module")
+def connection_list():
+    return [
+        {
+            "name": "test",
+        },
+    ]
+
+
+@pytest.fixture(scope="module")
+def connection_list_namespace():
+    return [
+        {
+            "name": "test",
+            "namespaces": ["test"]
+        },
+    ]
+
+
+@pytest.fixture(scope="module")
+def connection_with_base_values():
+    return [
+        {
+            "name": "test",
+            "namespaces": ["test"],
+            "use_service": True,
+            "create_groups": True,
+            "append_base_domain": True,
+            "base_domain": "test-domain"
+        },
+    ]
+
+
+@pytest.fixture(scope="module")
+def connection_with_network():
+    return [
+        {
+            "name": "test",
+            "namespaces": ["test"],
+            "use_service": True,
+            "create_groups": True,
+            "append_base_domain": True,
+            "base_domain": "test-domain",
+            "network_name": "test-network"
+        },
+    ]
+
+
+@pytest.fixture(scope="module")
+def connection_with_interface():
+    return [
+        {
+            "name": "test",
+            "namespaces": ["test"],
+            "use_service": True,
+            "create_groups": True,
+            "append_base_domain": True,
+            "base_domain": "test-domain",
+            "interface_name": "test-interface"
+        },
+    ]
+
+
+def vmi_options_base_values():
```

Comment:
I'm not sure how this can be done. In the way it is implemented is very convenient for asserting the GetVmiOptions() parameter of function call `get_vmi_for_namespace` :

```python
get_vmis_for_namespace.assert_called_once_with(mocker.ANY,
                                                   result["name"],
                                                   result["namespace"],
                                                   result["opts"])
```